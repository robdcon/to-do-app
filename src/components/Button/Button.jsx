import React from 'react';
import PropTypes from 'prop-types';
//import { Test } from './Button.styles';

const Button = (props) => (
  <div className="ButtonWrapper">
    Test content
  </div>
);

Button.propTypes = {
  // bla: PropTypes.string,
};

Button.defaultProps = {
  // bla: 'test',
};

export default Button;
