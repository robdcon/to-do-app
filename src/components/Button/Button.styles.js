import styled from 'styled-components';

export const StyledButton = styled.button`
box-shadow:inset 0px 1px 0px 0px #7a8eb9;
background:linear-gradient(to bottom, #637aad 5%, #5972a7 100%);
background-color:#637aad;
border:1px solid #314179;
display:inline-block;
cursor:pointer;
color:#ffffff;
font-family:Arial;
font-size:13px;
font-weight:bold;
padding:6px 12px;
text-decoration:none;
`;
