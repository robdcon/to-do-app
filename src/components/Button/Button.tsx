import React from 'react';
import { StyledButton } from './Button.styles';

interface Props {
  text?: string,
  type?:'submit',
  handleClick?: () => void,
  handleSubmit?: () => void
}

const Button:React.FC<Props> = (props) => (
  <div className="ButtonWrapper">
    <StyledButton data-testid="std-btn" onClick={props.handleClick} role="button">
      {props.text}
      {props.children}
    </StyledButton>
  </div>
);

export default Button;
