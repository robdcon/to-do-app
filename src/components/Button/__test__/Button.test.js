import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../Button';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'

afterEach(cleanup);

it("renders without crashing", () => {

    const div = document.createElement('div')
    ReactDOM.render(<Button></Button>, div)
})

it("renders button correctly", () => {

    const {getByTestId} = render(<Button text="Click Me"></Button>)
    expect(getByTestId('add-btn')).toHaveTextContent("Click Me")
})

it("renders button correctly again", () => {

    const {getByTestId} = render(<Button text="Click Me Again"></Button>)
    expect(getByTestId('add-btn')).toHaveTextContent("Click Me Again")
})

it("matches snapshot", () => {
    const tree = renderer.create(<Button text="save" />).toJSON();
    expect(tree).toMatchSnapshot();
})