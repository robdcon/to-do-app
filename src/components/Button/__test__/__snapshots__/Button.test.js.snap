// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ButtonWrapper"
  data-testid="std-btn"
>
  <button>
    save
  </button>
</div>
`;
